modules/models/src/main/java/org/eclipse/xpanse/modules/models/servicetemplate/AnsibleScriptConfig.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.modules.models.servicetemplate;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+import java.io.Serial;
+import java.io.Serializable;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * The config script of ansible.
+ */
+@Valid
+@Data
+@Slf4j
+public class AnsibleScriptConfig implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = 1L;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Name of the file in the script.")
+    private String playbookName;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Path to activate virtualenv.")
+    private String virtualEnv;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Version of the python.")
+    private float pythonVersion;
+
+    @NotNull
+    @Schema(description = "The agent will prepare the virtual environment if this true."
+            + "Otherwise it is assumed that the environment is already prepared "
+            + "as part of the resource creation.")
+    private Boolean isPrepareAnsibleEnvironment;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Url of the script repo.")
+    private String repoUrl;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Branch of the git repo.")
+    private String branch;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Full path of the requirements file in GIT repo.")
+    private String requirementsFile;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Install roles and collections.")
+    private String galaxyFile;
+
+    @NotNull
+    @NotBlank
+    @NotEmpty
+    @Schema(description = "Location of the playbook in repo.")
+    private String scriptsPath;
```

Comment:
i just realized that this parameter is not needed. We can merge this information with the `playbookName` parameter and delete this.